### PR TITLE
switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases

To allow users for smooth transitioning, keep also publishing to GCR.

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

File: hack/test-e2e-provider-local.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ cd $repo_root
 
 version=$(git rev-parse HEAD)
 make docker-images
-docker tag eu.gcr.io/gardener-project/gardener/extensions/networking-calico:latest networking-calico-local:$version
+docker tag europe-docker.pkg.dev/gardener-project/snapshots/extensions/networking-calico:latest networking-calico-local:$version
```
Comment (ScheererJ) [inline hack/test-e2e-provider-local.sh:33]: Will this work for release build? If we push them to a different path how should this work?

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Review by ScheererJ — CHANGES_REQUESTED** [review]
@ccwienk Please apply the same changes as in the cilium extension also here.

**Comment by ccwienk** [comment]
I think/hope to have addressed all review feedback (thanks for your thorough review!). This pull-request should now also be in sync w/ the one for [cilium](https://github.com/gardener/gardener-extension-networking-cilium/pull/231)

**Review by ScheererJ — APPROVED** [review]
Thanks for the changes.

/lgtm

**Pull request merged**